PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/rendering/networking/modelcontrollers/Requester.java

```diff
@@ -126,25 +125,22 @@ protected void getAdId() {
         }
 
         UserConsentManager userConsentManager = ManagersResolver.getInstance().getUserConsentManager();
+        WeakReference<Requester> weakRequester = new WeakReference<>(this);
         if (userConsentManager.canAccessDeviceData()) {
             AdIdManager.initAdId(context, new AdIdFetchListener() {
                 @Override
                 public void adIdFetchCompletion() {
                     LogUtil.info(TAG, "Advertising id was received");
-                    if (isDestroyed) {
-                        LogUtil.warning(TAG, "Attempted to make ad request after requester was destroyed");
-                    } else {
-                        makeAdRequest();
+                    if (weakRequester.get() != null) {
```

Comment:
This fix won't work. There is no guarantee that garbage collection will run after `Requester` gets destroyed.

Reply:
Yes, it makes sense. I missed that the `destroy()` method nullifies the `adResponseCallBack`. In this case, introducing the task for obtaining the adId looks more robust and suitable way. It will work in the same way as the `networkTask` in the same class. 

Or will it be overcoming and checking the references in the `sendAdException` enough?